### PR TITLE
Use make distcheck to provide tarball(s)

### PR DESCRIPTION
**Pull request by infirit**
Not sure how you tested this before..

The problem with the current approach is that it can easily miss including automake files that are in the dev package. For example the 2.3 tarball from the release job are broken as it does not include the gsettings automake parts.

ps: I uploaded 2 new files which pas distcheck.

### CODE REVIEW

**Review by cschramm — DISMISSED** [review]
I did not really get the gsettings issue, but this definitely makes sense. :+1: I updated the runners in #1780.

**Comment by infirit** [comment]
The gsettings automake parts are missing so we end up failing with

```
Making all in data
make[2]: Entering directory '/home/sander/Downloads/blueman-2.3/data'
Makefile:857: *** missing separator.  Stop.
make[2]: Leaving directory '/home/sander/Downloads/blueman-2.3/data'
make[1]: *** [Makefile:557: all-recursive] Error 1
make[1]: Leaving directory '/home/sander/Downloads/blueman-2.3'
make: *** [Makefile:468: all] Error 2
```

because of below in `Makefile`

```
# include the appropriate makefile rules for schema handling
@GSETTINGS_RULES@
```

File: configure.ac [hunk 1]
```diff
@@ -3,7 +3,7 @@ AC_PREREQ(2.61)
 AC_INIT([blueman], [2.3], [https://github.com/blueman-project/blueman/issues])
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIRS([m4])
-AM_INIT_AUTOMAKE([foreign dist-xz])
+AM_INIT_AUTOMAKE([1.16.4 foreign dist-xz])
```
[the remark this answers is not in the document]
Comment (cschramm) [reply configure.ac:6]: They don't. `mypy` and `test` do run on `ubuntu-22.04` but the only it does is host containers using the `python` docker images and those are based on Debian bullseye images. (It's pointless that I changed the host images :grimacing:)

[the remark this answers is not in the document]
Comment (infirit) [reply configure.ac:6]: Lol, I totally missed the docker image line. Ok I'll back this out for now.

**Review by cschramm — APPROVED** [review]

**Comment by cschramm** [comment]
Just to mention it somewhere (not too relevant in the light of #1793): The generated tarballs (releases 2.3 and 2.3.1) do not have proper meson support, as `meson_options.txt` is not included.

**Pull request merged**